lib/config-generator.js

```diff
@@ -155,67 +117,128 @@ export class ConfigGenerator {
+                importContent += "import pluginVue from \"eslint-plugin-vue\";\n";
+                exportContent += "  pluginVue.configs[\"flat/essential\"],\n";
+
+                // https://eslint.vuejs.org/user-guide/#how-to-use-a-custom-parser
+                if (useTs) {
+                    exportContent += "  { files: [\"**/*.vue\"], languageOptions: { parserOptions: { parser: tseslint.parser } } },\n";
+                }
+            }
+
+            if (this.answers.framework === "react") {
+                this.result.devDependencies.push("eslint-plugin-react");
+                importContent += "import pluginReact from \"eslint-plugin-react\";\n";
+                exportContent += "  pluginReact.configs.flat.recommended,\n";
+            }
 
-            // no need to install any plugin
-        } else if (this.answers.purpose === "problems") {
-            this.result.devDependencies.push("@eslint/js");
-            importContent += "import js from \"@eslint/js\";\n";
-            exportContent += `  { files: ["${extensions}"], plugins: { js }, extends: ["js/recommended"] },\n`;
+            if (needCompatHelper) {
+                this.result.devDependencies.push("@eslint/eslintrc", "@eslint/js");
+            }
+        } else {
+            exportContent += "  { ignores: [\"**/*.js\", \"**/*.cjs\", \"**/*.mjs\"] },\n";
         }
 
-        if (this.answers.language === "typescript") {
-            this.result.devDependencies.push("typescript-eslint");
-            importContent += "import tseslint from \"typescript-eslint\";\n";
-            exportContent += "  tseslint.configs.recommended,\n";
+        // language = json/jsonc/json5
+        if (languages.some(item => item.startsWith("json"))) {
+            this.result.devDependencies.push("@eslint/json");
+            importContent += "import json from \"@eslint/json\";\n";
+
+            if (languages.includes("json")) {
+                const config = purpose === "syntax"
+                    ? "  { files: [\"**/*.json\"], plugins: { json }, language: \"json/json\" },\n"
+                    : "  { files: [\"**/*.json\"], language: \"json/json\", extends: [\"json/recommended\"],\n";
+
+                exportContent += config;
+            }
+            if (languages.includes("jsonc")) {
+                const config = purpose === "syntax"
+                    ? "  { files: [\"**/*.jsonc\"], plugins: { json }, language: \"json/jsonc\" },\n"
+                    : "  { files: [\"**/*.jsonc\"], language: \"json/jsonc\", extends: [\"json/recommended\"] },\n";
+
+                exportContent += config;
+            }
+            if (languages.includes("json5")) {
+                const config = purpose === "syntax"
+                    ? "  { files: [\"**/*.json5\"], plugins: { json }, language: \"json/json5\" },\n"
+                    : "  { files: [\"**/*.json5\"], language: \"json/json5\", extends: [\"json/recommended\"] },\n";
```

Comment:
```suggestion
                    : "  { files: [\"**/*.json5\"], plugins: { json }, language: \"json/json5\", extends: [\"json/recommended\"] },\n";
```

We need to add the plugin here.